File: 1493-longest-subarray-of-1s-after-deleting-one-element/1493-longest-subarray-of-1s-after-deleting-one-element.py
class Solution(object):
    def longestSubarray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        i, zero_count, max_len = 0, 0, 0
        
        for j in range(len(nums)):
            if nums[j] == 0:
                zero_count += 1
            
            while 1 < zero_count:
                if nums[i] == 0:
                    zero_count -= 1
                i += 1
                
                
            max_len = max(max_len, j - i + 1)
            
        return max_len - 1
            

File: 1493-longest-subarray-of-1s-after-deleting-one-element/1493-longest-subarray-of-1s-after-deleting-one-element.rs
impl Solution {
    pub fn longest_subarray(nums: Vec<i32>) -> i32 {
        let(mut i, mut zero_cnt, mut max_len) = (0, 0, 0);
            for j in 0..nums.len() {
                if nums[j] == 0 {
                    zero_cnt += 1;
                }
        
                while zero_cnt > 1 {
                    if nums[i as usize] == 0 {
                        zero_cnt -= 1;
                    }
                    i += 1;
                }
                max_len = max_len.max((j as i32) - i + 1);
                
            }
            max_len - 1
    }
}